apps/api/migrations/subscribers/remove-duplicated-subscribers/remove-duplicated-subscribers.migration.ts

```diff
@@ -0,0 +1,150 @@
+import '../../../src/config';
+import { NestFactory } from '@nestjs/core';
+import { SubscriberRepository } from '@novu/dal';
+import { AppModule } from '../../../src/app.module';
+
+export async function removeDuplicatedSubscribers() {
+  // eslint-disable-next-line no-console
+  console.log('start migration - remove duplicated subscribers');
+
+  const app = await NestFactory.create(AppModule, {
+    logger: false,
+  });
+
+  const batchSize = 1000;
+  const subscriberRepository = app.get(SubscriberRepository);
+
+  const pipeline = [
+    // Group by subscriberId and _environmentId
+    {
+      $group: {
+        _id: { subscriberId: '$subscriberId', environmentId: '$_environmentId' },
+        count: { $sum: 1 },
+        subscribers: { $push: '$$ROOT' }, // Store all documents of each group
+      },
+    },
+    // Filter groups having more than one document (duplicates)
+    {
+      $match: {
+        count: { $gt: 1 },
+      },
+    },
+  ];
+
+  const cursor = await subscriberRepository._model.aggregate(pipeline, {
+    batchSize: batchSize,
+    readPreference: 'secondaryPreferred',
+    allowDiskUse: true,
+  });
+
+  for (const group of cursor) {
+    const { subscriberId, environmentId } = group._id;
+    const subscribers = group.subscribers;
+
+    if (subscribers.length <= 1) {
+      continue;
+    }
+
+    // sort oldest subscriber first
+    const sortedSubscribers = subscribers.sort((a, b) => a.updatedAt - b.updatedAt);
+    const mergedSubscriber = mergeSubscribers(sortedSubscribers);
+    const subscribersToRemove = sortedSubscribers.filter((subscriber) => subscriber._id !== mergedSubscriber._id);
+
+    // eslint-disable-next-line no-console
+    console.log(
+      'Merged subscriber:',
+      mergedSubscriber._id.toString(),
+      'subscriberId:',
+      subscriberId,
+      'environmentId:',
+      environmentId.toString()
+    );
+
+    try {
+      await subscriberRepository.update(
+        {
+          _id: mergedSubscriber._id,
+          subscriberId: subscriberId,
+          _environmentId: environmentId,
+        },
+        {
+          $set: mergedSubscriber,
```

Comment:
❓ **question:** shouldn't we just update some fields like `firstName, lastName, email, channels...` etc., without `_id, _environmentId, _organizationId` etc?